homeassistant/components/hive/__init__.py

```diff
@@ -69,75 +64,113 @@
         vol.Optional(ATTR_TIME_PERIOD, default="00:30:00"): vol.All(
             cv.time_period, cv.positive_timedelta, lambda td: td.total_seconds() // 60
         ),
-        vol.Required(ATTR_MODE): cv.string,
+        vol.Required(ATTR_ONOFF): cv.string,
     }
 )
 
 
 async def async_setup(hass, config):
-    """Set up the Hive Component."""
+    """Set up the Hive Integration."""
+    hass.data[DOMAIN] = {}
+
+    if DOMAIN not in config:
+        return True
+
+    conf = config[DOMAIN]
+
+    if not hass.config_entries.async_entries(DOMAIN):
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": config_entries.SOURCE_IMPORT},
+                data={
+                    CONF_USERNAME: conf[CONF_USERNAME],
+                    CONF_PASSWORD: conf[CONF_PASSWORD],
+                    CONF_SCAN_INTERVAL: conf.get(CONF_SCAN_INTERVAL, 120),
+                },
+            )
+        )
+    return True
+
+
+async def async_setup_entry(hass, entry):
+    """Set up Hive from a config entry."""
+    # Store an API object for your platforms to access
+    # hass.data[DOMAIN][entry.entry_id] = MyApi(...)
+
+    websession = aiohttp_client.async_get_clientsession(hass)
+    hive = Hive(websession)
+    hive_config = dict(entry.data)
+    hive_options = dict(entry.options)
 
     async def heating_boost(service):
         """Handle the service call."""
 
         entity_lookup = hass.data[DOMAIN]["entity_lookup"]
-        hive_id = entity_lookup.get(service.data[ATTR_ENTITY_ID])
-        if not hive_id:
+        device = entity_lookup.get(service.data[ATTR_ENTITY_ID])
+        if not device:
             # log or raise error
             _LOGGER.error("Cannot boost entity id entered")
             return
 
         minutes = service.data[ATTR_TIME_PERIOD]
         temperature = service.data[ATTR_TEMPERATURE]
 
-        hive.heating.turn_boost_on(hive_id, minutes, temperature)
+        await hive.heating.turn_boost_on(device, minutes, temperature)
 
     async def hot_water_boost(service):
         """Handle the service call."""
         entity_lookup = hass.data[DOMAIN]["entity_lookup"]
-        hive_id = entity_lookup.get(service.data[ATTR_ENTITY_ID])
-        if not hive_id:
+        device = entity_lookup.get(service.data[ATTR_ENTITY_ID])
+        if not device:
             # log or raise error
             _LOGGER.error("Cannot boost entity id entered")
             return
+
         minutes = service.data[ATTR_TIME_PERIOD]
-        mode = service.data[ATTR_MODE]
+        mode = service.data[ATTR_ONOFF]
 
         if mode == "on":
-            hive.hotwater.turn_boost_on(hive_id, minutes)
+            await hive.hotwater.turn_boost_on(device, minutes)
```

Comment:
How did this work before?

Reply:
it used to store the hive_id and pass that to the pyhiveapi library but now it stores the device against the entry_id and passed the to pyhiveapi.

Reply:
We haven't updated the library. How can this api change without a library update?

Reply:
Unfortunately this was an oversight in the last version, but is fixed now.